nat/traversal/pinger.go

```diff
@@ -304,44 +256,74 @@ func (p *Pinger) Valid() bool {
 	return true
 }
 
-func (p *Pinger) pingTargetConsumer(pingParams *Params) {
-	log.Info().Msgf("Pinging peer with: %+v", pingParams)
+func (p *Pinger) pingTargetConsumer(params Params) {
+	log.Info().Msgf("Pinging peer with: %+v", params)
 
-	if pingParams.ProxyPortMappingKey == "" {
+	if params.ProxyPortMappingKey == "" {
 		log.Error().Msg("Service proxy connection port mapping key is missing")
 		return
 	}
 
-	log.Info().Msgf("Ping target received: IP: %v, port: %v", pingParams.ConsumerPublicIP, pingParams.ConsumerPort)
-	if !p.natProxy.isAvailable(pingParams.ProxyPortMappingKey) {
-		log.Warn().Msgf("NATProxy is not available for this transport protocol key %v", pingParams.ProxyPortMappingKey)
+	stop := make(chan struct{})
+	defer close(stop)
+
+	conn, err := p.multiPing(params, 2, stop)
+	if err != nil {
+		log.Err(err).Msg("Failed to ping remote peer")
 		return
 	}
 
-	conn, err := p.getConnection(pingParams.ConsumerPublicIP, pingParams.ConsumerPort, pingParams.ProviderPort)
+	err = ipv4.NewConn(conn).SetTTL(128)
 	if err != nil {
-		log.Error().Err(err).Msg("Failed to get connection")
+		log.Error().Err(err).Msg("Failed to set connection TTL")
 		return
 	}
 
-	pingStop := make(chan struct{})
-	defer close(pingStop)
+	conn.Write([]byte("Using this connection"))
+
+	p.eventPublisher.Publish(event.AppTopicTraversal, event.BuildSuccessfulEvent(StageName))
+	log.Info().Msg("Ping received, waiting for a new connection")
+
+	go p.natProxy.handOff(params.ProxyPortMappingKey, conn)
+}
+
+func (p *Pinger) multiPing(params Params, initialTTL int, stop <-chan struct{}) (*net.UDPConn, error) {
+	if len(params.LocalPorts) != len(params.RemotePorts) {
+		return nil, errors.New("number of local and remote ports does not match")
+	}
+
+	type res struct {
+		conn *net.UDPConn
+		err  error
+	}
+
+	ch := make(chan res, len(params.LocalPorts))
+
+	for i := range params.LocalPorts {
+		go func(i int) {
+			conn, err := p.singlePing(params.IP, params.LocalPorts[i], params.RemotePorts[i], initialTTL+i, stop)
```

Comment:
Do I understand correctly what you assume same order between `RemotePorts` and `LocalPorts`. And each side has pre configured TTL. 
I suggest to make new structure configurable thru communication, TTLs pressented in tuples:
```
natPorts: [
  {local: 10001, remote: 20003, ttl: 124},
  {local: 10003, remote: 20001, ttl: 125},
]
```

Reply:
params is not a DTO. Local and remote ports are never transferred together. TTL does not need to be transferred.